ouroboros/src/ledger/mod.rs

```diff
@@ -0,0 +1,54 @@
+use mockall::automock;
+use pallas_crypto::hash::{Hash, Hasher};
+use thiserror::Error;
+
+#[derive(Error, Debug)]
+pub enum Error {
+    #[error("PoolId not found")]
+    PoolIdNotFound,
+}
+
+pub type PoolId = Hash<28>;
+
+/// The sigma value of a pool. This is a rational number that represents the total value of the
+/// delegated stake in the pool over the total value of the active stake in the network. This value
+/// is tracked in the ledger state and recorded as a snapshot value at each epoch.
+#[derive(Debug, PartialEq, Eq, Clone)]
+pub struct PoolSigma {
+    pub numerator: u64,
+    pub denominator: u64,
+}
+
+/// The pool info trait provides a lookup mechanism for pool data. This is sourced from the ledger
+#[automock]
+pub trait PoolInfo: Send + Sync {
+    /// Performs a lookup of a pool_id to its sigma value. This usually represents a different set of
+    /// sigma snapshot data depending on whether we need to look up the pool_id in the current epoch
+    /// or in the future.
+    fn sigma(&self, pool_id: &PoolId) -> Result<PoolSigma, Error>;
+
+    /// Hashes the vrf vkey of a pool.
+    fn vrf_vkey_hash(&self, pool_id: &PoolId) -> Result<[u8; 32], Error>;
+}
+
+/// The node's cold vkey is hashed with blake2b224 to create the pool id
+pub fn issuer_vkey_to_pool_id(issuer_vkey: &[u8]) -> PoolId {
```

Comment:
```suggestion
pub fn issuer_vkey_to_pool_id(issuer_vkey: &[u8; 32]) -> PoolId {
```

Reply:
Perhaps a good idea long term to introduce some type alias and a generic interface around crypto capabilities? That's one of the things I believe the Haskell codebase does quite right. The crypto interface is fully captured and bundled as a coherent set of primitives: https://github.com/IntersectMBO/cardano-base/tree/master/cardano-crypto-class/src/Cardano/Crypto

I believe this _could be_ `pallas-crypto` in the long run.

Reply:
I'll make a type alias for this in pallas-crypto.

Reply:
Converted this to just `Bytes` from pallas-code which is what the `issuer_vkey` is when it is extracted from the block cbor.